### PR TITLE
Add dp & port metrics for failure/logoff/success.

### PR DESCRIPTION
**Pull request by Bairdo**
Add handlers for failure/logoff - not implemented on chewie side.
Add partial test for logoff - counters not implemented on chewie side.
Add option to connect to wpa_supplicant's control socket.
Also corrected indentation of the last pr.

### CODE REVIEW

File: faucet/faucet_dot1x.py [hunk 1]
```diff
@@ -56,12 +56,30 @@ def auth_handler(self, address, _group_address):
         self.logger.info(
             'Successful auth from MAC %s on %s' % (
                 str(address), self.dot1x_port))
-        self._valve._inc_var('dot1x_success') # pylint: disable=protected-access
+        self.metrics.dp_dot1x_success.labels(**self._valve.base_prom_labels).inc()
```
Comment (Bairdo) [inline faucet/faucet_dot1x.py]: Not sure how these should be done (protected-access). (_inc_var looks tidier)

File: faucet/faucet_dot1x.py [hunk 2]
```diff
@@ -56,12 +56,30 @@ def auth_handler(self, address, _group_address):
...
+        self.metrics.port_dot1x_success.labels(**self._valve._port_labels(self.dot1x_port)).inc() # pylint: disable=protected-access
```
Comment (Bairdo) [inline faucet/faucet_dot1x.py]: Also not sure about this protected access.

**Comment by Bairdo** [comment]
ugh I've mucked the rebase from master up.

**Comment by Bairdo** [comment]
I think this is right now, but i'll check tomorrow and possibly do a squash.

File: faucet/faucet_dot1x.py (hunk 1 above)
Comment (KitL) [reply faucet/faucet_dot1x.py]: maybe part of the functionality of _inc_var should be moved to a function in the metrics object to make this a little cleaner. But I think it is a little weird for dot1x to ask valve to update its metrics for it.

**Comment by Bairdo** [comment]
Ok, fixed the rebase issue and squashed.

File: faucet/faucet_dot1x.py (hunk 2 above)
Comment (Bairdo) [reply faucet/faucet_dot1x.py]: thanks renamed in  commit 8c142e7

File: faucet/faucet_dot1x.py (hunk 1 above)
Comment (Bairdo) [reply faucet/faucet_dot1x.py]: moved part of _inc_var the metrics object in commit 8dc8c9c

**Pull request merged**